core/server/worker/src/main/java/alluxio/worker/netty/AbstractReadHandler.java

```diff
@@ -317,27 +316,28 @@ protected boolean acceptMessage(Object object) {
   }
 
   /**
-   * Initializes the handler for the given block read request.
-   *
    * @param request the block read request
+   * @return an instance of read request based on the request read from channel
    */
-  protected abstract void initializeRequest(Protocol.ReadRequest request) throws Exception;
+  protected abstract T createRequest(Protocol.ReadRequest request) throws Exception;
 
   /**
-   * Returns the appropriate {@link DataBuffer} representing the data to send, depending on the
-   * configurable transfer type.
-   *
-   * @param channel the netty channel
-   * @param len The length, in bytes, of the data to read from the block
-   * @return a {@link DataBuffer} representing the data
+   * Completes the read request. When the request is closed, we should clean up any temporary state
```

Comment:
Is this the correct comment? The implementation of `create...` is supposed to complete/close/clean something?